Microsoft.Toolkit.Uwp.SampleApp/SamplePages/InAppNotification/InAppNotificationCode.bind

```diff
@@ -0,0 +1,85 @@
+// ## XAML - Text + buttons (DataTemplate)
+
+<DataTemplate x:Key="InAppNotificationWithButtonsTemplate">
+    <UserControl>
+        <Grid DataContext="{Binding ElementName=CurrentPage}">
+            <VisualStateManager.VisualStateGroups>
+                <VisualStateGroup>
+                    <VisualState x:Name="NarrowState">
+                        <VisualState.Setters>
+                            <Setter Target="TextBlock.(Grid.RowSpan)" Value="1" />
+                            <Setter Target="TextBlock.(Grid.ColumnSpan)" Value="2" />
+
+                            <Setter Target="ButtonsStackPanel.(Grid.Row)" Value="1" />
+                            <Setter Target="ButtonsStackPanel.(Grid.RowSpan)" Value="1" />
+                            <Setter Target="ButtonsStackPanel.(Grid.Column)" Value="0" />
+                            <Setter Target="ButtonsStackPanel.(Grid.ColumnSpan)" Value="2" />
+                            <Setter Target="ButtonsStackPanel.Orientation" Value="Vertical" />
+                            <Setter Target="ButtonsStackPanel.Margin" Value="0 5 0 0" />
+
+                            <Setter Target="YesButton.Width" Value="Auto" />
+                            <Setter Target="YesButton.HorizontalAlignment" Value="Stretch" />
+
+                            <Setter Target="NoButton.Width" Value="Auto" />
+                            <Setter Target="NoButton.HorizontalAlignment" Value="Stretch" />
+                            <Setter Target="NoButton.Margin" Value="0 5 0 0" />
+                        </VisualState.Setters>
+                                    
+                        <VisualState.StateTriggers>
+                            <StateTrigger IsActive="{Binding IsRootGridActualWidthLargerThan700, Converter={StaticResource BoolNegationConverter}, Mode=OneWay}" />
+                        </VisualState.StateTriggers>
+                    </VisualState>
+
+                    <VisualState x:Name="LargeState">
+                        <VisualState.StateTriggers>
+                            <StateTrigger IsActive="{Binding IsRootGridActualWidthLargerThan700, Mode=OneWay}" />
+                        </VisualState.StateTriggers>
+                    </VisualState>
+                </VisualStateGroup>
+            </VisualStateManager.VisualStateGroups>
+
+            <Grid x:Name="RootGrid" SizeChanged="RootGrid_SizeChanged">
```

Comment:
Ok. I remember. I made it that way to be able to detect the change of the Grid Width (not the App Width) so I can create a StateTrigger based on the container Size, not the App Size.

I do not know if there is a better option because it will crash if you do not have this code in your code-behind: 

```
public bool IsRootGridActualWidthLargerThan700 { get; set; }

private void RootGrid_SizeChanged(object sender, SizeChangedEventArgs e)
{
    // When the root part size of the In App Notification template changed, we should apply VisualState
    bool newValue = e.NewSize.Width > 700;

    if (IsRootGridActualWidthLargerThan700 != newValue)
    {
        IsRootGridActualWidthLargerThan700 = newValue;
        OnPropertyChanged(nameof(IsRootGridActualWidthLargerThan700));
    }
}
```